app/Http/Controllers/UserController.php

```diff
@@ -103,4 +103,23 @@ public function choosePlan(Request $request, $planId)
     return response()->json(['message' => 'Plan chosen successfully', 'user' => $user]);
 }
 
+public function requestPasswordReset(Request $request)
+{
+    $data = $request->validate([
+        'email' => 'required|email',
+    ]);
+
+    return $this->userService->requestPasswordReset($data['email']);
+}
+
+public function resetPassword(Request $request)
+{
+    $data = $request->validate([
+        'token' => 'required|string',
+        'newPassword' => 'required|string|max:255',
```

Comment:
as for the password you should use this type of validation
'password' => 'required|string|confirmed|min:6',
when we have confirmed the user is asked to confirm the password by writing it again like this
  "password": "Isra123!",
    "password_confirmation": "Isra123!",